### DISCUSSION TITLE
Unable to do UUID compare in CCC function

### DISCUSSION THREAD

**Discussion by cgbdevelco** [zephyrproject-rtos/zephyr#41030] [Q&A]
Hello,
I have created different characteristics with the CCC macro (BT_GATT_CCC) that share the same "Configuration changed callback" function.

Inside the function below, i try to get the UUID of the characteristic to determine which one notifications has been enabled on.

static void cccCfgChanged(const struct bt_gatt_attr *attr, uint16_t value)
{
    char str[BT_UUID_STR_LEN];
    bt_uuid_to_str(attr->uuid, str, sizeof(str));
}
The function bt_uuid_to_str just give junk in the char array, does attr-uuid not contain the uuid of the characteristic which has enabled notifications?
I have also tried  bt_uuid_cmp without any luck.

**Comment by Thalley** [comment]
@cgbdevelco Can you post what value you get in `str`?

I just tried your code, and I get 2902 (`BT_UUID_GATT_CCC_VAL`) as expected.

**Reply by cgbdevelco** [reply]
Maybe I just misunderstood and thought I could get the UUID of the characteristic which are linked with the CCC. Anyway I have solved this by using the bt_gatt_is_subscribed to check before using bt_gatt_notify instead of setting a flag,

**Reply by Thalley** [reply]
I am not aware of any way to get the value characteristic from the CCC. Perhaps @Vudentz knows if it is possible, but looks like it isn't. 

The attr->user_data is supposedly a `struct _bt_gatt_ccc`, so we don't have a reference from the `attr` pointer at all.
